tdp/core/dag.py

```diff
@@ -274,28 +275,29 @@ def warning(collection_name: str, message: str) -> None:
         logger.warning(message + f", collection: {collection_name}")
 
     for operation_name, operation in nodes.items():
-        c_warning = functools.partial(warning, operation.collection_name)
-        for dependency in operation.depends_on:
-            # *_start operations can only be required from within its own service
-            dependency_service = nodes[dependency].service_name
-            if (
-                dependency.endswith("_start")
-                and dependency_service != operation.service_name
-            ):
-                c_warning(
-                    f"Operation '{operation_name}' is in service '{operation.service_name}', depends on "
-                    f"'{dependency}' which is a start action in service '{dependency_service}' and should "
-                    f"only depends on start action within its own service"
-                )
+        if operation.collection_name is not None:
```

Comment:
`collection_name` is optional for an operation ?

Reply:
At the moment yes. In the class `Operation`:

```
def __init__(
        self,
        name: str,
        collection_name: Optional[str] = None,
```

Reply:
Maybe you could invert the condition (using `continue`) to improve code and diff.

Reply:
I understand you want to fix red marker but these marker remind us that the code need refactor. With this fix we will not remember to remove this `if` when `collection_name` will be mandatory.